src/averageAges.js

```diff
@@ -36,8 +42,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+
+function calculateWomenAverageAge(people, withChildren = false) {
+  const women = people.filter(person => withChildren
+    ? people.find(child => person.name === child.mother && person.sex === 'f')
+    : person.sex === 'f');
+
+  return +(sumOfAges(women)
+    / women.length).toFixed(2);
+}
+
+function sumOfAges(people) {
```

Comment:
```suggestion
function calculateSumOfAges(people) {
```
![image](https://user-images.githubusercontent.com/96409563/235341483-25cb597e-dd4f-45dc-af6c-feacf55afce4.png)